srcDoc/tutorials/performance_validation.rst

```diff
@@ -0,0 +1,635 @@
+to one switch. The first thing we will need to know is what is the *minimum latency* and *maximum throughput*
+achievable in that environment, independent on using DDS or not. This means that we need to know
+how fast can we send with a simple UDP-socket communication. Luckily this is something that we can get with
+**Perftest**: By using the ``-rawTransport`` option, we skip the use of *RTPS* and *DDS* and we
+just send using UDPv4 sockets.
+
+We will be doing a *Latency Test* and a *Throughput Test* (See
+`this <https://github.com/rticommunity/rtiperftest/blob/3.0/srcDoc/introduction.rst#latency-test-vs-throughput-test>`__ section to understand the
+differences).
+
+Once that is done, we will have a baseline which is going to tell us the minimum latency we can expect
+and the maximum throughput achievable in the system, even when not using *RTPS* and *DDS*. The next step
+is then to execute **Perftest** compiled for *Professional* and for *Micro* and see the equivalent results.
+
+UDPv4 Communication (Raw Transport)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Throughput Test
+---------------
+
+The maximum throughput of this scenario will be limited by several factors: If the size of the samples we
+are sending is small, the CPU consumption will be high, since it will need to iterate through the process
+of sending the samples to the NIC quite often. If the size of the sample is big enough, then the problem
+is the physical limitations of the network itself, how fast the nics and the switch are.
+
+In our specific case, the switch is a 1Gbps switch which should not be the cap, since the *Raspberry Pi* nics
+we are using have 100Mbps NICs. Then, 100Mbps is our maximum theoretical throughput.
+
+Given all this information, the right way to perform the test is by iterating through different data sizes. We
+will use the following commands:
+
+* **Publisher side**
+
+    .. code::
+
+        for DATALEN in 32 64 128 256 512 1024 2048 4096 8192 16384 32768 63000; do
+            bin/armv6vfphLinux3.xgcc4.7.2/release/perftest_cpp -pub -peer 10.45.3.119 -nic eth0 -raw -pub -noPrint -exec 20 -datalen $DATALEN -batchSize 0;
+        done
+
+* **Subscriber side**
+
+    .. code::
+
+        for DATALEN in 32 64 128 256 512 1024 2048 4096 8192 16384 32768 63000; do
+            bin/armv6vfphLinux3.xgcc4.7.2/release/perftest_cpp -sub -peer 10.45.3.120 -nic eth0 -raw -noPrint -datalen $DATALEN;
+        done
+
+Some comments about the parameters we used:
+
+* In `Raw Transport Mode` the `-scan` option is not available, that is why we need to iterate through
+  the different data sizes using a for loop (in `bash`).
+
+* In `Raw Transport Mode` we do not have a discovery mechanism, as we do have when
+  Using **RTI Connext DDS**, therefore, it is required to use the `-peer` parameter.
+
+* In throughput mode, by default **Perftest** uses "batching", we want to disable it for this
```

Comment:
Ok, I see your point. But this is artificial: Batching is a feature that we have in DDS, and we are "exporting" it to Raw Transport, but still it remains as a "trick", is a clever way to get more throughput, but it is not native to sockets. Not sure if I explain myself: You want to compare again sending data in the most crude (raw) way: I take the data I create the packet and I send it via UDPv4 sockets. I don't do fancy stuff like keeping a vector and then sending them all together

Reply:
Once could say: Oh, so I can implement batching with Raw Data and I have it. Yes... ok, enjoy your time doing it...  But what i mean is that would be something else than just send using sockets, it would be a "customized" solution.

Reply:
This is all questionable because we are not comparing apples to apples. Doing batching with sockets is not highly complex. Also, what happens if the user use the command line option -batch with raw transport. Is that allowed?

Reply:
In fact I also see above this:

In throughput mode, by default Perftest uses "batching", we want to disable it for this test, that can be done by using -batchSize 0.

This does not make sense. We are telling the user don't do batching with sockets. We do we even expose the option?

Reply:
I modified this sentence so it makes more sense:

```
* In throughput mode, by default, **Perftest** uses "batching", as this feature is not
  native to sending using sockets, we have implemented it at the application level
  in the **Perftest** application. Therefore, in order to compare the raw transport behavior,
  we want to disable it for this test, that can be done simply by using `-batchSize 0`.
```